### PR TITLE
[WIP] PT 1.7.1 Support for 0.3.0 Release + Merge to master from PT 0.2.1 release

### PR DESCRIPTION
**Pull request by dhanainme**
Updates from 0.2.1 - Naked dir support
PT 1.7.1 , PT vision / text / audio upgrades

### CODE REVIEW

File: ts_scripts/install_utils [hunk 1]
```diff
@@ -28,9 +28,9 @@ install_java_deps()
```
Comment (maaquib) [inline ts_scripts/install_utils:28]: I see this being used in only two [places](https://github.com/pytorch/serve/search?q=%22install_utils%22). Do we still need this or can we migrate it to one of the python util scripts

**Review by maaquib — CHANGES_REQUESTED** [review]
@harshbafna @dhanainme 
```
## DLAMI 38, g4dn.xl, ami-02e86b825fe559330
sudo rm /usr/local/cuda && sudo ln -s /usr/local/cuda-11.0 /usr/local/cuda
ll /usr/local/cuda

git clone https://github.com/pytorch/serve/
cd serve/
git checkout pt_171_support_ts_03
python ts_scripts/install_dependencies.py --environment dev --cuda=latest
python torchserve_sanity.py
python test/regression_tests.py
```

Doesn't seem to be installing the correct version with `--cuda=latest` flag
```
...
Collecting torch==1.6.0
  Downloading torch-1.6.0-cp37-cp37m-manylinux1_x86_64.whl (748.8 MB)
     |████████████████████████████████| 748.8 MB 104.2 MB/s eta 0:00:01
...
```

**Comment by harshbafna** [comment]
@maaquib: PFB below the logs from testing on cuda 11.0 environment 



```
ubuntu@ip-172-31-68-243:~$ git clone https://github.com/pytorch/serve.git
Cloning into 'serve'...
remote: Enumerating objects: 137, done.
remote: Counting objects: 100% (137/137), done.
remote: Compressing objects: 100% (118/118), done.
remote: Total 15931 (delta 79), reused 29 (delta 19), pack-reused 15794
Receiving objects: 100% (15931/15931), 40.01 MiB | 54.42 MiB/s, done.
Resolving deltas: 100% (9039/9039), done.
ubuntu@ip-172-31-68-243:~$ cd serve
ubuntu@ip-172-31-68-243:~/serve$ git checkout pt_171_support_ts_03
Branch 'pt_171_support_ts_03' set up to track remote branch 'pt_171_support_ts_03' from 'origin'.
Switched to a new branch 'pt_171_support_ts_03'
ubuntu@ip-172-31-68-243:~/serve$ python ts_scripts/install_dependencies.py --environment dev
<snip install_dependency logs>
ubuntu@ip-172-31-68-243:~/serve$ nvcc --version
nvcc: NVIDIA (R) Cuda compiler driver
Copyright (c) 2005-2020 NVIDIA Corporation
Built on Wed_Jul_22_19:09:09_PDT_2020
Cuda compilation tools, release 11.0, V11.0.221
Build cuda_11.0_bu.TC445_37.28845127_0
ubuntu@ip-172-31-68-243:~/serve$ pip show torch
Name: torch
Version: 1.7.1
Summary: Tensors and Dynamic neural networks in Python with strong GPU acceleration
Home-page: https://pytorch.org/
Author: PyTorch Team
Author-email: packages@pytorch.org
License: BSD-3
Location: /home/ubuntu/anaconda3/lib/python3.7/site-packages
Requires: numpy, typing-extensions
Required-by: torchvision, torchtext, torchaudio, captum
ubuntu@ip-172-31-68-243:~/serve$ python
Python 3.7.6 (default, Jan  8 2020, 19:59:22) 
[GCC 7.3.0] :: Anaconda, Inc. on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import torch
>>> torch.cuda.is_available()
True
>>>

File: ts_scripts/install_utils (hunk 1 above)
Comment (harshbafna) [reply ts_scripts/install_utils:28]: This has been referenced in Circle CI. I think we can move out the circle ci related code and this script to a different PR. This file doesn't affect the current installation scripts. No changes required in existing python utils.

**Review by harshbafna — APPROVED** [review]

**Review by maaquib — APPROVED** [review]

**Pull request merged**